### ISSUE TITLE
External Packages Not Available in Spack

### ISSUE THREAD

**Issue by citibeth** [spack/spack#7971]
There have been reports that if somebody wishes to use an external (OS-installed) package whose version is not available in Spack's corresponding `packages.yaml`, then this doesn't work.  Making this work requires adding a new version to the Spack package to correspond to the OS-installed version.  See #7961, for example.

This should not be required.

**Comment by alalazo** [comment]
Uhm, I installed `openssl@system` on our clusters a 100 times today (well, Jenkins did):
```console
$ spack install openssl%gcc
==> openssl@system : externally installed in /usr/lib64
==> openssl@system : generating module file
==> openssl@system : registering into DB

$ spack info openssl
Package:   openssl

Description:
    OpenSSL is an open source project that provides a robust, commercial-
    grade, and full-featured toolkit for the Transport Layer Security (TLS)
    and Secure Sockets Layer (SSL) protocols. It is also a general-purpose
    cryptography library.

Homepage: http://www.openssl.org

Tags: 
    None

Preferred version:  
    1.0.2n    http://www.openssl.org/source/openssl-1.0.2n.tar.gz

Safe versions:  
    1.1.0g    http://www.openssl.org/source/openssl-1.1.0g.tar.gz
    1.1.0e    http://www.openssl.org/source/openssl-1.1.0e.tar.gz
    1.1.0d    http://www.openssl.org/source/openssl-1.1.0d.tar.gz
    1.1.0c    http://www.openssl.org/source/openssl-1.1.0c.tar.gz
    1.0.2n    http://www.openssl.org/source/openssl-1.0.2n.tar.gz
    1.0.2m    http://www.openssl.org/source/openssl-1.0.2m.tar.gz
    1.0.2k    http://www.openssl.org/source/openssl-1.0.2k.tar.gz
    1.0.2j    http://www.openssl.org/source/openssl-1.0.2j.tar.gz
    1.0.2i    http://www.openssl.org/source/openssl-1.0.2i.tar.gz
    1.0.2h    http://www.openssl.org/source/openssl-1.0.2h.tar.gz
    1.0.2g    http://www.openssl.org/source/openssl-1.0.2g.tar.gz
    1.0.2f    http://www.openssl.org/source/openssl-1.0.2f.tar.gz
    1.0.2e    http://www.openssl.org/source/openssl-1.0.2e.tar.gz
    1.0.2d    http://www.openssl.org/source/openssl-1.0.2d.tar.gz
    1.0.1u    http://www.openssl.org/source/openssl-1.0.1u.tar.gz
    1.0.1t    http://www.openssl.org/source/openssl-1.0.1t.tar.gz
    1.0.1r    http://www.openssl.org/source/openssl-1.0.1r.tar.gz
    1.0.1h    http://www.openssl.org/source/openssl-1.0.1h.tar.gz

Variants:
    Name [Default]      Allowed values    Description


    systemcerts [on]    True, False       Use system certificates

Installation Phases:
    install

Build Dependencies:
    zlib

Link Dependencies:
    zlib

Run Dependencies:
    None

Virtual Packages: 
    None
```

**Comment by alalazo** [comment]
Marking `unreproducible` until somebody finds a reproducer that triggers the error.

**Comment by citibeth** [comment]
Maybe the bug comes up if you try to install a numbered version, rather
than "system".

**Comment by adamjstewart** [comment]
It does not come up if you try to install a numbered version, see https://github.com/spack/spack/pull/7961#issuecomment-385969796

If no one knows of any way to reproduce this, it is by definition "unreproducible".

**Comment by citibeth** [comment]
> If no one knows of any way to reproduce this, it is by definition "unreproducible".

Let's please wait for @goxberry to weigh in before jumping to that conclusion.  He has just submitted 3 new-version PRs, so clearly he must be experiencing something.

**Comment by citibeth** [comment]
Maybe it comes up if you're trying to install something that depends on your external package.  @goxberry can you describe your experience in more detail?

**Comment by adamjstewart** [comment]
> Maybe it comes up if you're trying to install something that depends on your external package.

It doesn't, see https://github.com/spack/spack/pull/7961#issuecomment-385969796

You also submitted several new-version PRs for this back in the day, but you never reported how to reproduce this.

**Comment by citibeth** [comment]
> You also submitted several new-version PRs for this back in the day, but you never reported how to reproduce this.

Sorry, I didn't take the time at that point to make that report.  Too many other things were going on with me and Spack at that time, and this issue fell low on the priority.  But now is 1+year later, and things are a little more stable.

Maybe @goxberry is using an old version of Spack and this has been fixed since then?  I really don't know.  In any case, I'm reluctant to take the lead on issues that don't affect me; given that there are some BIG things that DO effect me, that will take a concerted effort to get merged.

**Comment by citibeth** [comment]
@goxberry Are you able to submit bug details to this Issue?  If not, we will unfortunately have to close it for now.

**Comment by citibeth** [comment]
I'm pretty sure this has been fixed in recent versions of Spack.  I've verified that old versions of packages I had previous added are no longer needed.  See:
https://github.com/spack/spack/pull/8009

**Comment by citibeth** [comment]
It looks like we have our specific example of failure in #8046

**Comment by adamjstewart** [comment]
Can you see if this is fixed by #3161?
